backend/src/modules/doubles-tournament/core/domain/MatchesQueue.ts

```diff
@@ -28,11 +30,19 @@ export class MatchesQueue {
     return [...this.queue];
   }
 
-  withStatusChangedMatch(match: QueuedMatch): MatchesQueue {
-    const queue = this.queuedMatches.filter((queuedMatch) => queuedMatch.matchNumber.raw !== match.matchNumber.raw);
+  withStatusChangedMatch(matchNumber: MatchNumber, status: MatchStatus, tableNumber?: number): MatchesQueue {
+    const queue = this.queuedMatches.map((queuedMatch) => {
+      if (queuedMatch.matchNumber.raw === matchNumber.raw) {
+        queuedMatch.status = status;
+        if (tableNumber) {
+          queuedMatch.tableNumber = tableNumber;
```

Comment:
Skoro to funkcja map to nie powinismy miec side effectów. Bo map bierze jedno i zwraca drugie. 
Czyli zwróć nowy obiekt z tableNumber nowym.
```
{
...queuedMatch,
tableNumber: queuedMatch.tableNumber ?? tableNumber
}
```
I kodzik tez bardziej zweizly :)